lib.php

```diff
@@ -115,8 +126,30 @@ function local_smartmedia_pluginfile($course, $cm, $context, $filearea, $args, $
         return false; // Source file doesn't match smart file.
     }
 
-    // TODO: add check to make sure user can access source file. (MDL-66006).
+    // Setup cache for content.
+    $cachekey = sha1($filename . sesskey());
```

Comment:
this cache key feel like it doesn't have enough info to stop a lot of clashes

Reply:
I'd like to see a test with two files with the same name but in different course sections / course, eg intro.mpg and see if it gets confused between them

Reply:
The filename is extremely distinct, its already a hash, collisions here are very hard